src/hb-ot-shaper-arabic.cc

```diff
@@ -232,10 +229,13 @@ collect_features_arabic (hb_ot_shape_planner_t *plan)
   if (plan->props.script == HB_SCRIPT_ARABIC)
     map->add_gsub_pause (arabic_fallback_shape);
 
-  /* No pause after rclt.  See 98460779bae19e4d64d29461ff154b3527bf8420. */
-  map->enable_feature (HB_TAG('r','c','l','t'), F_MANUAL_ZWJ);
-  map->enable_feature (HB_TAG('c','a','l','t'), F_MANUAL_ZWJ);
-  map->add_gsub_pause (nullptr);
+   map->enable_feature (HB_TAG('c','a','l','t'), F_MANUAL_ZWJ);
+   /* https://github.com/harfbuzz/harfbuzz/issues/1573 */
+   if (!map->has_feature (HB_TAG('r','c','l','t')))
```

Comment:
This should check for `calt`, NOT `rclt`, humm? The code as written doesn't make sense: it only adds rclt if the font does not have rclt!! I think this essentially removed rclt from Arabic? @khaledhosny

Reply:
`rclt` is still added by `hb_ot_shape_collect_features()` since it is included in `horizontal_features`:
https://github.com/harfbuzz/harfbuzz/blob/a070f9ebbe88dc71b248af9731dd49ec93f4e6e6/src/hb-ot-shape.cc#L298-L308

I think the effect of this change is adding a pause after `calt` only if there is no `rclt` in the font. Which is essentially what you said in: https://github.com/harfbuzz/harfbuzz/pull/3724#issuecomment-1184801987

I think the `map->enable_feature (HB_TAG('r','c','l','t'), F_MANUAL_ZWJ);` is superflous and should be dropped, it only confuses things.

Reply:
Thanks. Dropped it.